### ISSUE TITLE
Consider deploying to dedicated namespace?

### ISSUE THREAD

**Issue by scaleoutsean** [ThinkParQ/beegfs-csi-driver#6]
I used the "one liner" to deploy, it deployed to `kube-system`. 

Perhaps we should document that this approach doesn't deploy to a dedicated namespace and suggest to create one?

**Comment by ejweber** [comment]
I agree that we should use a dedicated namespace. When we added OpenShift and operator support, we made sure that the default in these environments was to deploy into the beegfs-csi namespace (and suggest to create it if it didn't already exist).

I'd like to go back in time and set up our standard deployment manifests to deploy the driver to the beegfs-csi namespace automatically as well. Changing the default namespace now, however, creates upgrade issues. If a user initially deployed to the `kube-system` namespace and uses a manifest that references the `beegfs-csi` namespace to upgrade, they'll actually end up with two drivers; an old one in `kube-system` and a new one in `beegfs-csi`. Of course, we could document this change in our patch notes, but our philosophy has generally been to avoid any additional upgrade steps when possible.

It would be easy enough to simply add the note you are suggesting around the one liner. If you (@scaleoutsean) and the team think this is "enough", we can get it in for the next release. Whether or not we should change the default deployment manifests themselves is a bit larger of a discussion. Thoughts?

**Comment by scaleoutsean** [comment]
I think it's enough to recommend. 

I don't know how the modern Trident CSI does it (I don't use Helm, I still use the old CLI way of installing it which installs to default namespace if I don't specify it with `-n).  The Trident docs ask the user to stick to the same method for installing and uninstalling the driver as mixing installation procedures did cause problems in the past (I don't remember why, maybe Helm defaults to best practices and uses "trident", but doesn't check if there's existing driver in another NS or something like that), so it seems it's very hard make it completely foolproof.

**Comment by ejweber** [comment]
I though about this some more as I was getting ready to make the documentation change and I changed my mind. We currently specify kube-system in the default kustomize overlay: 

https://github.com/NetApp/beegfs-csi-driver/blob/316c1cdac57365ab39c56aabad4354c153eb8579/deploy/k8s/overlays/default/kustomization.yaml#L6

But we document in multiple locations that you should create a custom overlay by copying the default overlay and use the custom overlay for upgrades. For example:

https://github.com/NetApp/beegfs-csi-driver/blob/316c1cdac57365ab39c56aabad4354c153eb8579/docs/deployment.md?plain=1#L57-L59

https://github.com/NetApp/beegfs-csi-driver/blob/316c1cdac57365ab39c56aabad4354c153eb8579/deploy/k8s/README.md?plain=1#L19-L46

In short, I think it is safe to change the default deployment namespace to beegfs-csi and to rely on our users having correctly installed the driver using an overlay if they expected to maintain it over the life of multiple versions. Thoughts?

**Comment by ejweber** [comment]
This change will be in the upcoming v1.2.2 release. I'll note the commit hash and close this issue when it is publicly available.

**Comment by ejweber** [comment]
This change was in https://github.com/NetApp/beegfs-csi-driver/commit/81c234a7d3a3b798bdca16dc34291492f8cfcd4e.
